db/system_keyspace.cc

```diff
@@ -1503,6 +1503,27 @@ future<> system_keyspace::cache_truncation_record() {
     });
 }
 
+    future<> system_keyspace::drop_truncation_rp_records() {
+        sstring req = format("SELECT table_uuid, shard, segment_id from system.{}", TRUNCATED);
+        auto rs = co_await execute_cql(req);
+
+        bool any = false;
+        co_await parallel_for_each(rs->begin(), rs->end(), [&] (const cql3::untyped_result_set_row& row) -> future<> {
+            auto table_uuid = table_id(row.get_as<utils::UUID>("table_uuid"));
+            auto shard = row.get_as<int32_t>("shard");
+            auto segment_id = row.get_as<int64_t>("segment_id");
+
+            if (segment_id != 0) {
+                any = true;
+                sstring req = format("UPDATE system.{} SET segment_id = 0, position = 0 WHERE table_uuid = {} AND shard = {}", TRUNCATED, table_uuid, shard);
```

Comment:
why set to 0 instead of set to null / delete?
(is there a difference?)

From commit message I understand you need the static column to still be set, but you can DELETE chosen columns  (and I'm not sure if clustering row deletion also applies to the static column? if not you could just delete the row)